src/hammer-tech/schema.json

```diff
@@ -194,7 +194,43 @@
             "type" : "string"
           }
         },
-        "additionalProperties" : false
+        "additionalproperties" : false
+      }
+    },
+    "stackup" : {
+      "type" : "array",
+      "items" : {
+        "title" : "Metal",
+        "type" : "object",
+        "properties" : {
+          "name" : {
+            "type" : "string"
+          },
+          "index" : {
+            "type" : "integer"
+          },
+          "direction" : {
+            "type" : "string"
+          },
+          "min_width" : {
+            "type" : "number"
+          },
+          "pitch" : {
+            "type" : "number"
+          },
+          "offset" : {
+            "type" : "number"
+          },
+          "spacingtable_widths" : {
```

Comment:
If this is meant to model the following LEF info, maybe we can have spacingtable be an array of tuples instead?
```
SPACINGTABLE ORTHOGONAL 
WITHIN 0.15 SPACING 0.11
WITHIN 0.13 SPACING 0.13
WITHIN 0.11 SPACING 0.15 
```

Reply:
Hmm. Maybe, let me sit on this. I think that'll be more verbose than I'd like, but this is not a strong opinion.

Reply:
I think my decision would be made by whichever makes the code easier to decipher.

Reply:
From a data structure perspective I would prefer the list of tuples as it better represents the structure in the LEF, unless I am mistaken there

Reply:
True, the list sizes should be the same but that should be an error imo. I think the example you listed is a degenerate case, though- it's usually a multidimensional structure and we've distilled it into a simpler case by sacrificing area efficiency a bit by assuming infinite run length.

Reply:
Would the following structure be more future proof then?

```python
class SpacingEntry(NamedTuple):
  # if width < than this
  width: float
  # if length < than this
  length: Optional[float]
  # then follow this spacing
  spacing: float
```

We could also be agile about it and not include the `length` field until we need it

Reply:
I don't see us ever needing this info, TBH. I only included it so we can calculate the power strap widths, but even that is a bit of a micro-optimization IMO. I would rather go with this and continue making progress on stuff that matters, this is getting to be bikeshedding a bit.

Reply:
I suppose if we are going to include it we may as well do it in an extensible way. If it's extra work/burden for you I can help with this.

Reply:
I think it's a waste of time right now. We can change it later. Feel free to file an issue.

Reply:
Resolved offline: file an issue and deal with API breakages later